### PR TITLE
Added topic_arn config to SNS events

### PR DESCRIPTION
**Pull request by eahefnawy**
Added functionality for SNS events to accept a `topic_arn` configuration, which enables the function to hook to a pre-existing SNS topic. Resolves Issue #1607

cc/ @flomotlik @pmuens

### CODE REVIEW

**Comment by pmuens** [comment]
Good one! Thanks for that!

Could you also add a brief documentation to the `docs/guide/overview-of-event-sources` doc? Thanks!

**Comment by pmuens** [comment]
BTW. Have you tested it?

**Comment by flomotlik** [comment]
Please also describe steps to validate this so its easy to do on our end as well

**Comment by pmuens** [comment]
Tested it locally.



Unfortunately with no success (the lambda has no trigger for the SNS topic with the topic ARN).

Here are the steps to test it:

1. Create a new SNS topic in the AWS console
2. Create a SNS event for a function in `serverless.yml`
3. Paste in the ARN of the previously created SNS topic (see below)
4. Deploy

```
functions:
  hello:
    handler: handler.hello
    events:
      - sns: my-sns-topic
  world:
    handler: handler.world
    events:
      - sns: arn:aws:sns:us-east-1:123456:yet-another-topic
```

The first topic is created and connected to the `hello` function. The second `world` function is not connected to the SNS topic `yet-another-topic`.

Maybe I'm missing something...

**Comment by flomotlik** [comment]
@pmuens could you put the CF template in a Gist and post here?

**Comment by eahefnawy** [comment]
@pmuens when you created the topic on aws console, did you also subscribe for the function?

The issue is CF doesn't have a way to just create a subscription on an existing topic. With CF, you can only add subscriptions array when you create the topic using the `AWS::SNS:Topic` template. 

So if the user choses to create the topic manually, he has to subscribe to the function himself. Then the framework can take care of the second step which is giving this topic permission to invoke that lambda, and that's what this PR does.

**Comment by pmuens** [comment]
@eahefnawy Ok. That was the issue.

It works after I've subscribed the lambda function to the SNS topic (but this is really poor UX, although there's no way around it).

/cc @flomotlik

**Comment by pmuens** [comment]
I've successfully tested it (as mentioned above).

Here are the steps to test it:

Here are the steps to test it:

1. Create a new SNS topic in the AWS console
2. Create a SNS event for a function in `serverless.yml`
3. Paste in the ARN of the previously created SNS topic (see below)
4. Deploy
5. Go to the SNS topic, click edit and subscribe to the lambda function

Done

```
functions:
  hello:
    handler: handler.hello
    events:
      - sns: my-sns-topic
  world:
    handler: handler.world
    events:
      - sns: arn:aws:sns:us-east-1:123456:yet-another-topic
```

It's GTM from my side once the `docs` are added to the `docs/guide/overview-of-event-sources` section.

**Comment by eahefnawy** [comment]
@pmuens one final note is that step 5 could be done anytime (doesn't have to be the last step). The function just need to be deployed so you can find it in the list. But yeah I agree it's bad UX and not super sure many users will use it. I'd personally just have Serverless take care of creating everything from scratch.

**Comment by flomotlik** [comment]
@eahefnawy please update the docs accordingly, otherwise people will have no idea that they have to do that

**Comment by eahefnawy** [comment]
@pmuens @flomotlik updated the docs according to your comments 💃

File: docs/guide/overview-of-event-sources.md [hunk 1]
```diff
@@ -131,6 +131,15 @@ functions:
     events:
       - sns: dispatch
 ```
+Or if you have a pre-existing topic ARN, you can just provide the topic ARN instead:
```
Comment (pmuens) [inline docs/guide/overview-of-event-sources.md]: Could you add the hint that the user should subscribe the function to the lambda here as well? Thanks!

**Comment by pmuens** [comment]
@eahefnawy thanks for that! Added one tiny comment. After that it's GTM from my side :shipit: 

Good job!

File: docs/guide/overview-of-event-sources.md (hunk 1 above)
Comment (eahefnawy) [inline docs/guide/overview-of-event-sources.md]: ok done. However I don't see why we're duplicating docs in multiple places! The overview is meant to give a quick start (ie. I don't see the extra options for APIG in the overview) and to ignore the details. This is an overall docs issue though :)

**Comment by pmuens** [comment]
Awesome! GTM :shipit:

**Comment by pmuens** [comment]
@eahefnawy just one tiny question:

Why do we use `topic_arn` instead of `topicArn` as we're always using `camelCase` in our `serverless.yml`. I would prefer to stick to one style if possible...

**Comment by eahefnawy** [comment]
Because we already have `topic_name` and `display_name` instead of `topicName` and `displayName`:)

**Comment by pmuens** [comment]
Yep. That makes sense. I've opened up another Issue (see: https://github.com/serverless/serverless/issues/1804) so that we later refactor it and make everything `camelCase`. But we should do this in another PR.

So GTM! Great job @eahefnawy 💃  :shipit: 🚢

**Pull request merged**